primitives/finality-grandpa/src/lib.rs

```diff
@@ -370,25 +373,32 @@ where
 /// Localizes the message to the given set and round and signs the payload.
 #[cfg(feature = "std")]
 pub fn sign_message<H, N>(
+	keystore: BareCryptoStorePtr,
 	message: grandpa::Message<H, N>,
-	pair: &AuthorityPair,
+	public: &AuthorityId,
 	round: RoundNumber,
 	set_id: SetId,
-) -> grandpa::SignedMessage<H, N, AuthoritySignature, AuthorityId>
+) -> Result<grandpa::SignedMessage<H, N, AuthoritySignature, AuthorityId>, ()>
```

Comment:
() error type is unconventional. What semantic are you trying to convey?

Reply:
I followed the convention of the functions in that same file where error returned is `()`. I was fine with this as when calling this function `sign_message`, any error i can see there would be caused by the keystore so when calling that function i map it into an `sp_consensus::Error`.

Reply:
Than you should use `Option`.

Reply:
Sorry I don't know what was wrong with me to declare those functions as `Result<(), ()>` rather than `Option<()>`.

Reply:
Semantically I'd prefer result if we are indicating that something has gone wrong. And there is good precedent in leaking as little information as possible about a cryptographic error, as is done in Ring. But it should be a new struct, not `()`.

Reply:
I opted to use Option here... unless @rphmeier you have strong opinion about using `Result` with an actual error?